### PR TITLE
fix background color style

### PR DESCRIPTION
**Pull request by JerryWu1234**
image

- [ ] I've read and followed the [contributing guide](https://github.com/mui/mui-toolpad/blob/master/CONTRIBUTING.md#sending-a-pull-request) on how to create great pull requests.
- [ ] I've updated the relevant documentation for any new or updated feature
- [ ] I've linked relevant GitHub issue with "Closes #<issue id>"
- [ ] I've added a visual demonstration under the form of a screenshot or video

### CODE REVIEW

**Comment by JerryWu1234** [comment]
if the inset was set, background was #f7f7f7, I think the preview mode follows the dev

**Comment by Janpot** [comment]
I believe the preview is correct, it's the canvas that overlays a slight transparent grey when nothing is selected. The idea is to emphasize something is selected by slightly dimming the rest. It gives the illusion of a grey background.

We can start by disabling this overlay when nothing is selected. If this turns out to still be confusing we will disable it altogether and rely purely on the blue box to indicate selection

**Comment by JerryWu1234** [comment]
>  we will disable it altogether and rely purely on the blue box to indicate selection
I chose this way

**Comment by JerryWu1234** [comment]
> > we will disable it altogether and rely purely on the blue box to indicate selection
> 
> I chose this way image image

@Janpot 
It only shows the blue button active, which can help shrink the code that doesn’t need to calculate it a lot.

**Comment by Janpot** [comment]
If we don't want a background at all, you may as well remove this whole component

**Comment by JerryWu1234** [comment]
> If we don't want a background at all, you may as well remove this whole component



> If we don't want a background at all, you may as well remove this whole component

actually, I removed the whole component but the `[chromium] › integration/rest-basic/index.spec.ts:36:5 › rest runtime basics ` this test will occur an error.
let me dive into deeper

**Comment by JerryWu1234** [comment]
image

**Comment by Janpot** [comment]
Yes, you may have to move the `data-testid="page-overlay"` one comonent up

**Comment by JerryWu1234** [comment]
@Janpot 
after my in-depth study, 
We can’t maybe directly delete this part of PinholeOverlay, because it is not only for emphasizing something that is selected, but also for canceling the status of active when something is selected.

**Comment by JerryWu1234** [comment]
I added a test for the status of active

**Comment by apedroferreira** [comment]
If the PinholeOverlay does have utility and the only issue was the darkening of the UI I think just removing that darkening should do!
Thanks for adding a test for this, I'll leave some comments on it.

File: test/integration/editor/index.spec.ts [hunk 1]
```diff
@@ -115,3 +115,20 @@ test('code editor auto-complete', async ({ page }) => {
   await page.keyboard.type('textF');
   await expect(page.getByRole('option', { name: 'textField' })).toBeVisible();
 });
+
+test('cancel the status of active when click the white space', async ({ page }) => {
```
Comment (apedroferreira) [inline test/integration/editor/index.spec.ts:119]: I'd reword the test name to something like `must deselect selected element when clicking outside of it`

File: test/integration/editor/index.spec.ts [hunk 2]
```diff
@@ -115,3 +115,20 @@ test('code editor auto-complete', async ({ page }) => {
...
+
+  await editorModel.goToPageById('K7SkzhT');
+
+  await editorModel.waitForOverlay();
+
+  const text = editorModel.appCanvas.getByText('text-foo');
+
+  await clickCenter(page, text);
+  await expect(editorModel.appCanvas.getByTestId('node-hud-tag')).toBeVisible();
+  const element = await editorModel.appCanvas.locator('.Toolpad_HudOverlay');
```
Comment (apedroferreira) [inline test/integration/editor/index.spec.ts:130]: Instead of having a locator for the overlay and using `clickCenter` I think that here we should write a `page.mouse.click` with coordinates outside the text element that is selected.

So we can get `text.boundingBox()` for the text element, and then click on `textBoundingBox. x + testBoundingBox.width + 50` and `textBoundingBox.y + testBoundingBox.height + 50`, for example. This way we are sure that we are clicking outside of it.

File: test/integration/editor/index.spec.ts [hunk 3]
```diff
@@ -115,3 +115,20 @@ test('code editor auto-complete', async ({ page }) => {
...
+  await clickCenter(page, element);
+  const hudButton1 = await editorModel.appCanvas.getByTestId('node-hud-tag');
```
Comment (apedroferreira) [inline test/integration/editor/index.spec.ts:132]: No need to write this locator again, you just have to write it once above and then reuse it.
Something like ```const textNodeHudTag = editorModel.appCanvas.getByTestId('node-hud-tag')```

**Review by apedroferreira — CHANGES_REQUESTED** [review]
Adding some suggestions for the test, other than that this approach seems to work!

**Comment by JerryWu1234** [comment]
@apedroferreira 
help agree on this pr if this pr can be merged.

File: test/integration/editor/index.spec.ts [hunk 4]
```diff
@@ -192,3 +192,18 @@ test('code editor auto-complete', async ({ page }) => {
...
 });
+
+test('must deselect selected element when clicking outside of it', async ({ page }) => {
+  const editorModel = new ToolpadEditor(page);
+
+  await editorModel.goToPageById('K7SkzhT');
+
+  await editorModel.waitForOverlay();
+  const nodeHudTag = editorModel.appCanvas.getByTestId('node-hud-tag');
+  const input = editorModel.appCanvas.locator('input:nth-child(1)');
```
Comment (apedroferreira) [inline test/integration/editor/index.spec.ts:203]: Do we need to use `nth-child`? Or if there's only one single input in the page, can we make it even more simple?

Comment (apedroferreira) [inline test/integration/editor/index.spec.ts:203]: The reason I said we can use a text field was to make it full-width, so that the selection and the element have the same size.
Can you make the `fullWidth` prop be `true` for this text field?

File: test/integration/editor/index.spec.ts [hunk 5]
```diff
@@ -192,3 +192,18 @@ test('code editor auto-complete', async ({ page }) => {
...
   await expect(page.getByRole('option', { name: 'textField' })).toBeVisible();
 });
+
+test('must deselect selected element when clicking outside of it', async ({ page }) => {
+  const editorModel = new ToolpadEditor(page);
+
+  await editorModel.goToPageById('K7SkzhT');
+
+  await editorModel.waitForOverlay();
+  const nodeHudTag = editorModel.appCanvas.getByTestId('node-hud-tag');
```
Comment (apedroferreira) [inline test/integration/editor/index.spec.ts:202]: If would be even better if this node HUD selector could only match the HUD of the `TextField`, but not the HUD of the `Text`.

**Review by apedroferreira — CHANGES_REQUESTED** [review]
Almost there, just a few more important changes in the test!
Also just a nitpick, but feel free to use more vertical space in the test to separate bits of logic, it can really help improve the readability!

File: test/integration/editor/index.spec.ts (hunk 5 above)
Comment (JerryWu1234) [reply test/integration/editor/index.spec.ts:202]: I'm not very good at the `playwright`,
Could you share your idea with me?
thanks

File: test/integration/editor/index.spec.ts [hunk 6]
```diff
@@ -192,3 +192,18 @@ test('code editor auto-complete', async ({ page }) => {
...
+  const input = editorModel.appCanvas.locator('input').nth(0);
```
Comment (apedroferreira) [inline test/integration/editor/index.spec.ts:203]: There's only one input in this page so you can just do:
```
  const input = editorModel.appCanvas.locator('input');
```

**Comment by apedroferreira** [comment]
With my comments above, here's how I would vertically space the test, order the variables and change variable names to improve readability and make the test as easy to understand as possible:
```
const editorModel = new ToolpadEditor(page);

  await editorModel.goToPageById('K7SkzhT');

  await editorModel.waitForOverlay();

  const textField = editorModel.appCanvas.locator('input');
  const textFieldNodeHudTag = editorModel.appCanvas
    .getByTestId('node-hud-tag')
    .filter({ hasText: 'textField' });

  await clickCenter(page, textField);
  await expect(textFieldNodeHudTag).toBeVisible();

  const textFieldBoundingBox = await textField.boundingBox();
  await page.mouse.click(textFieldBoundingBox!.x + 150, textFieldBoundingBox!.y + 150);
  await expect(textFieldNodeHudTag).toBeHidden();
  ```

File: test/integration/editor/index.spec.ts (hunk 5 above)
Comment (JerryWu1234) [reply test/integration/editor/index.spec.ts]: thanks.
I got it wrong, I thought I needed to modify the tsx file before

**Review by Armen692244 — COMMENTED** [review]
J
